skglm/utils.py

```diff
@@ -393,6 +393,18 @@ def check_group_compatible(obj):
             )
 
 
+def alpha_max_group_lasso(X, y, n_groups, grp_indices, grp_ptr, weights):
```

Comment:
IMO no need to implement a function for alpha_max as so far, we use it only one time.

Reply:
so far yes, but we might use it more than once in the future. So I added it in `utils` as suggested by @mathurinm .

Reply:
yes it's legit. just make it private to avoid maintenance costs